app/modules/wallet/addresses/addresses.route.unit.spec.ts

```diff
@@ -0,0 +1,196 @@
+        sandbox
+          .mock(WalletAddress)
+          .expects('findAll')
+          .resolves(null);
+        supertest(app)
+          .get('/api/wallets/5/addresses')
+          .end((err: any, res: supertest.Response) => {
+            if (err) {
+              done(err);
+            } else {
+              expect(res.status).to.equal(404);
+              done();
+            }
+          });
+      });
+
+      it('should throw 500 if server error while reading db', (done) => {
+        sandbox
+          .mock(WalletAddress)
+          .expects('findAll')
+          .throws(null);
+        supertest(app)
+          .get('/api/wallets/5/addresses')
+          .end((err: any, res: supertest.Response) => {
+            if (err) {
+              done(err);
+            } else {
+              expect(res.status).to.equal(500);
+              done();
+            }
+          });
+      });
+    });
+    describe('PUT /api/wallets/:id/addresses', () => {
+      const addressObj = {
+        chain: 'eth',
+        address: '1x',
+        name: 'eth address'
+      };
+      const addressReturn = {
+        walletAddress: {
+          id: 3,
+          chain: 'eth',
+          address: '1x',
+          name: 'eth address',
+          wallet_id: 1,
+          updated_at: '2018-08-02T07:00:49.026Z',
+          created_at: '2018-08-02T07:00:49.026Z'
+        }
+      };
+
+      it('should save wallet addresses successfully', (done) => {
+        sandbox
+          .mock(WalletAddress)
+          .expects('create')
+          .resolves(addressReturn);
+        supertest(app)
+          .put('/api/wallets/1/addresses')
+          .send(addressObj)
+          .end((err: any, res: supertest.Response) => {
+            if (err) {
+              done(err);
+            } else {
+              expect(res.status).to.equal(200);
+              done();
+            }
+          });
+      });
+
+      it('should throw 500 if server error while writing db', (done) => {
+        sandbox
+          .mock(WalletAddress)
+          .expects('create')
+          .throws(null);
+        supertest(app)
+          .put('/api/wallets/1/addresses')
+          .send(addressReturn)
+          .end((err: any, res: supertest.Response) => {
+            if (err) {
+              done(err);
+            } else {
+              expect(500);
+              done();
+            }
+          });
+      });
+    });
+
+    describe('DELETE /api/wallets/:id/addresses/:addressid', () => {
+      it('should delete wallet address successfully', (done) => {
+        sandbox
+          .mock(WalletAddress)
+          .expects('destroy')
+          .resolves({
+            message: 'Deleted Address'
+          });
+        supertest(app)
+          .put('/api/wallets/1/addresses/1')
+          .send()
+          .end((err: any, res: supertest.Response) => {
+            if (err) {
+              done(err);
+            } else {
+              expect(200);
```

Comment:
Also add exactly what to expect the response. So if anything to that response is changed. its test fails.